src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -762,19 +793,25 @@ namespace winrt::TerminalApp::implementation
         // Add an event handler when the terminal wants to paste data from the Clipboard.
         term.PasteFromClipboard({ this, &TerminalPage::_PasteFromClipboardHandler });
 
+        // Bind Tab events to the TermControl and the Tab's Pane
+        hostingTab->BindEventHandlers(term);
```

Comment:
We are constructing `Tab`s through `make_shared<Tab>`, so it didn't make sense at first why it was throwing [`bad_weak_ptr`](https://en.cppreference.com/w/cpp/memory/bad_weak_ptr). But after some digging, I got a theory 🕵. Since all [`shared_ptr` constructors](https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr) don't construct the managed object directly. `make_shared` at some point must call `::new T`, and at this moment `T` won't have the context that it will become a managed object. So at the time of `Tab`'s construction we can't call and weaken `shared_from_this`. Look the try/catch section of the example [from cppreference](https://en.cppreference.com/w/cpp/memory/enable_shared_from_this). I think it demonstrates our issue here.

Reply:
We are constructing `Tab`s through `make_shared<Tab>`, so it didn't make sense at first why it was throwing [`bad_weak_ptr`](https://en.cppreference.com/w/cpp/memory/bad_weak_ptr). But after some digging, I got a theory 🕵. Since all `shared_ptr` constructors don't construct the managed object directly. `make_shared` at some point must call `::new T`, and at this moment `T` won't have the context that it will become a managed object. So at the time of `Tab`'s construction we can't call and weaken `shared_from_this`. Look the try/catch section of the example [from cppreference](https://en.cppreference.com/w/cpp/memory/enable_shared_from_this). I think it demonstrates our issue here.

Reply:
Oh. Well I learned something new, thanks!